source/WeightedList.h

```diff
@@ -48,9 +48,9 @@ class WeightedList {
 	// Average the result of the given function by the choices' weights.
 	template <class Callable>
 	typename std::enable_if<
-		std::is_arithmetic<typename std::result_of<Callable&&(const Type&&)>::type>::value,
+		std::is_arithmetic<typename std::invoke_result<Callable&&, const Type&&>::type>::value,
 		// The return type of WeightedList::Average, if the above test passes:
-		typename std::result_of<Callable&&(const Type&&)>::type
+		typename std::invoke_result<Callable&&, const Type&&>::type
```

Comment:
Can you switch to using the `_t`/`_v` versions while you're at it? So like `typename std::enable_if<...>::type` => `std::enable_if_t<...>`. In other places too

Reply:
~~Uhh... how do I do that?~~

Reply:
Remove `typename` and `::type` and add a `_t` to the name. For values remove `::value` and add `_v`

Reply:
I _might_ have also replaced every other ::value and ::type as well.